Simulator/player.py

```diff
@@ -277,81 +281,89 @@ def findItem(self, name):
-                                # This should update the item name section of the vector
-                                for z in range(6, 0, -1):
-                                    if i_index > 2 * z:
-                                        input_array[11 + 6 * (i + 1) - z] = 1
-                                        i_index -= 2 * z
-                lower_bound = 29 * (x + 7 * y)
-                output_array[lower_bound: lower_bound + 29] = input_array
-        self.board_vector = output_array
-        self.generate_player_vector()
+                    occupation_state = 1
+                else:
+                    occupation_state = 0
+                # Can use another conditional to indicate that the unit should be sold
+                output_array[hex_count] = [x,y,occupation_state]
+                hex_count += 1
+        self.board_occupation_vector = output_array.reshape(28*3)
+        self.generate_player_vector() # Not sure why player vector is called here
+        self.generate_champion_vectors()
+
+    def generate_champion_vectors(self):
+        '''
+        Helps to generate the vectors containing all the champions information. Including both Board and Bench
+        :return:
+        '''
+        output_array = np.zeros((self.MAX_CHAMPION,self.CHAMPION_INFORMATION)) # 19 * 11 = 209
+
+        #Check champion on bench from self.bench
+        curr_bench_count = 0
+        for x in range(0, 9):
+            champion_info_array = np.zeros(self.CHAMPION_INFORMATION)
+            if self.bench[x]:
+                self.generate_single_champion_vector(self.bench[x], champion_info_array)
+                output_array[curr_bench_count] = champion_info_array
+
+        #Check champion on board from self.board
+        curr_board_count = 9 # Starts at 9, skipping the 9 spot on bench
+        for x in range(0, 7):
+            for y in range(0, 4):
+                champion_info_array = np.zeros(self.CHAMPION_INFORMATION)
+                if self.board[x][y]:
+                    self.generate_single_champion_vector(self.board[x][y], champion_info_array)
+                    output_array[curr_board_count] = champion_info_array
+                    curr_board_count += 1
+        self.champions_owned_vector = output_array.reshape(self.MAX_CHAMPION * self.CHAMPION_INFORMATION)
+    def generate_single_champion_vector(self, curr_champ, champion_info_array):
+        '''
+        Helps to generate a vector of length CHAMPION_INFO
+        :param curr_champ: Object champion
+        :param champion_info_array: The array to update the information in
+        :return:
+        '''
+        # start with champion name
+        c_index = list(COST.keys()).index(
+            curr_champ.name) + 1  # Returns index of champion, # Avoiding index 0 as index 0 is reserved for no chammpion
+        # This should update the champion name section of the vector
+        champion_info_array[0] = float(c_index) / self.MAX_CHAMPION_IN_SET

```

Comment:
You are loading the COST array every time for this index. Save that array somewhere at the start. We call this way too often for us to warrant loading it in every time.